docs/draft/17-url-data.md

```diff
@@ -0,0 +1,126 @@
+---
+permalink: /draft/17
+title: 17/URL data
+parent: Draft specs
+layout: default
+---
+
+# 17/URL data
+
+> Version: 0.1
+>
+> Status: Draft
+>
+> Authors: Felicio Mununga <felicio@status.im>
+
+## Abstract
+
+This document describes serialization, compression and encoding of data within URLs, so the app links could carry enough information to help users with instant previewing and verifying even prior clicking or taking an action on the content.
+
+## Related scope
+
+### Specs
+
+- See <https://github.com/status-im/specs/pull/159> for URL scheme
+
+### Features
+
+- Onboarding website
+- Link preview
+- Link sharing
+- Deep linking
+- Routing and navigation
+
+### App entities
+
+- Community
+- Channel
+- User
+
+## Requirements
+
+### Product
+
+- Encode data within URL until Waku response is instant and considered reliable enough for link previewing and visiting of the onboarding website. Or until a product roadmap for alternative decentralized storage is finalized.
+- A verify state while loading onboarding website until encoded data is matched against Waku response to mitigate identity attacks
+- Accept only `[A-Za-z0-9_-.\u0020]` character class for textual fields
+- Shortest result possible
+
+## Implementation
+
+### Example
+
+- See <https://github.com/status-im/status-web/pull/345/files>
+
+### Data
+
+#### Fields
+
+```protobuf
+syntax = "proto3";
+
+message Community {
+ string display_name = 1;
+ string description = 2;
+ uint32 members_count = 3;
+ string color = 4;
+}
+
+message Channel {
+ string display_name = 1;
+ string description = 2;
+ string emoji = 3;
+ string color = 4;
+ Community community = 5;
+}
+
+message User {
+ string display_name = 1;
+ string description = 2;
+ string color = 3;
+}
+
+message URLData {
+ // Community, Channel, or User
+ bytes content = 1;
+}
+
+// Field on CommunityDescription, CommunityChat and ContactCodeAdvertisement
+message URLProps {
+ string encoded_url_data = 1;
```

Comment:
ℹ️ Stores the url data already encoded alongside the other proto messages so clients don't have to do the additional computation for other members/communities, just mapping now.